recipes/dd-agent.rb

```diff
@@ -102,12 +102,13 @@ def template_vars # rubocop:disable Metrics/AbcSize
 end
 
 # Common configuration
-service_provider = nil
 if node['datadog']['agent6'] &&
-   (((node['platform'] == 'amazon' || node['platform_family'] == 'amazon') && node['platform_version'].to_i != 2) ||
-   (node['platform'] != 'amazon' && node['platform_family'] == 'rhel' && node['platform_version'].to_i < 7))
+  (((node['platform'] == 'amazon' || node['platform_family'] == 'amazon') && node['platform_version'].to_i != 2) ||
+  (node['platform'] != 'amazon' && node['platform_family'] == 'rhel' && node['platform_version'].to_i < 7))
   # use Upstart provider explicitly for Agent 6 on Amazon Linux < 2.0 and RHEL < 7
   service_provider = Chef::Provider::Service::Upstart
+else
```

Comment:
this `else` block will also include Agent 5, but with Agent 5 we should never hardcode the `service_provider` (Agent 5 generally only provides sysv-init scripts, and relies on the init system's compatibility layer with sysv-init scripts)